### PR TITLE
Added controllers, jsp files for different operations with Car, Driver, Manufacturer

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/core/basesyntax/controller/AddDriverToCarController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long driverId = Long.valueOf(req.getParameter("driver_id"));
+        Long carId = Long.valueOf(req.getParameter("car_id"));
+        Car car = carService.get(carId);
+        Driver driver = driverService.get(driverId);
+        carService.addDriverToCar(driver, car);
+        resp.sendRedirect(req.getContextPath() + "/car/all");
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/controller/AddDriverToCarController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/car/");
```
just a suggestion

File: src/main/java/core/basesyntax/controller/AllCarsController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import core.basesyntax.service.CarService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AllCarsController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("core.basesyntax");
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/controller/AllCarsController.java]:
```suggestion
    private static final Injector INJECTOR= Injector.getInstance("core.basesyntax");
```
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html#:~:text=Class%20names%20should%20be%20nouns,such%20as%20URL%20or%20HTML

File: src/main/java/core/basesyntax/controller/AllCarsController.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import core.basesyntax.model.Car;
+import core.basesyntax.service.CarService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AllCarsController extends HttpServlet {
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/controller/AllCarsController.java]:
```suggestion
public class GetAllCarsController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/car/creation.jsp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Creation</title>
+</head>
+<body>
+<h1>Please, provide cars details</h1>
+<form method="post" action="${pageContext.request.contextPath}/car/creation">
+    Please, fill car`s model: <input type="text" name="model">
+    and manufacturer id: <input type="text" name="manufacturer_id">
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/views/car/creation.jsp]: what if there is no manufacturer with your input id? and what if you input "2hk23ds5" here?

File: src/main/webapp/WEB-INF/views/car/driverToCar.jsp [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Add driver to car</title>
+</head>
+<body>
+<h1>Please, provide car and driver id</h1>
+<form method="post" action="${pageContext.request.contextPath}/car/driverToCar">
+    Please, enter car id: <input type="text" name="car_id">
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/views/car/driverToCar.jsp]: what if your input will be blank?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/controller/AllCarsController.java (hunk 2 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/AllCarsController.java]: Fixed

File: src/main/java/core/basesyntax/controller/AllCarsController.java (hunk 3 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/AllCarsController.java]: Fixed

File: src/main/java/core/basesyntax/controller/AddDriverToCarController.java (hunk 1 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/AddDriverToCarController.java]: The idea was that the user can see what he added) but maybe you right - maybe he may want to add some drivers and after looks for the result

File: src/main/webapp/WEB-INF/views/car/creation.jsp (hunk 4 above)
Comment (vicire) [reply src/main/webapp/WEB-INF/views/car/creation.jsp]: Fixed

File: src/main/webapp/WEB-INF/views/car/driverToCar.jsp (hunk 5 above)
Comment (vicire) [reply src/main/webapp/WEB-INF/views/car/driverToCar.jsp]: Fixed

File: src/main/java/core/basesyntax/controller/CreateCarController.java [hunk 6]
```diff
@@ -25,10 +26,24 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+                req.getRequestDispatcher("/WEB-INF/views/car/creation.jsp").forward(req, resp);
+            }
+        } catch (NumberFormatException e) {
+            req.setAttribute("message", "Please, enter numeric id");
+            req.getRequestDispatcher("/WEB-INF/views/car/creation.jsp").forward(req, resp);
+        } catch (NoSuchElementException e) {
+            req.setAttribute("message", "There is no such manufacturer id, "
+                    + "please enter valid id");
+            req.getRequestDispatcher("/WEB-INF/views/car/creation.jsp").forward(req, resp);
+        }
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/controller/CreateCarController.java]: google `required` and `type="number"`

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/core/basesyntax/controller/CreateCarController.java (hunk 6 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/CreateCarController.java]: 👍

File: src/main/java/core/basesyntax/controller/AddDriverToCarController.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+public class AddDriverToCarController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("core.basesyntax");
+    private final CarService carService = (CarService) INJECTOR.getInstance(CarService.class);
+    private final DriverService driverService = (DriverService) INJECTOR
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/driverToCar.jsp").forward(req, resp);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/controller/AddDriverToCarController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/car/addDriver.jsp").forward(req, resp);
```

File: src/main/java/core/basesyntax/controller/AddDriverToCarController.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            Long driverId = Long.valueOf(req.getParameter("driver_id"));
+            Long carId = Long.valueOf(req.getParameter("car_id"));
+            Car car = carService.get(carId);
+            Driver driver = driverService.get(driverId);
+            carService.addDriverToCar(driver, car);
+            resp.sendRedirect(req.getContextPath() + "/car/");
+        } catch (NoSuchElementException e) {
+            req.setAttribute("message", "There is no such id, please enter valid id");
+            req.getRequestDispatcher("/WEB-INF/views/car/driverToCar.jsp").forward(req, resp);
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/controller/AddDriverToCarController.java]: let's not catch RuntimeExceptions. Make all the validations from the jsp side. In scope of this task we may just "believe" that we will get the existing id

File: src/main/java/core/basesyntax/controller/CreateCarController.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+public class CreateCarController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("core.basesyntax");
+    private final CarService carService = (CarService) INJECTOR.getInstance(CarService.class);
+    private final ManufacturerService manufacturerService = (ManufacturerService) INJECTOR
+            .getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/creation.jsp").forward(req, resp);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/controller/CreateCarController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/car/creat.jsp").forward(req, resp);
```

File: src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java [hunk 10]
```diff
@@ -99,7 +99,7 @@ public Car create(Car car) {
     public Car update(Car car) {
         String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
                 + "WHERE car_id = ? AND deleted = FALSE";
-        String deleteQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String deleteQuery = "DELETE FROM cars_drivers WHERE car_id = ? AND driver_id IS NOT NULL";
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java]: this should be handle by constraint in the table for this field

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>AllDrivers</servlet-name>
+        <servlet-class>core.basesyntax.controller.GetAllDriversController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>AllDrivers</servlet-name>
+        <url-pattern>/driver/all</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers</url-pattern>
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>CreateDriver</servlet-name>
+        <servlet-class>core.basesyntax.controller.CreateDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>CreateDriver</servlet-name>
+        <url-pattern>/driver/creation</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/creat</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+    <servlet>
+        <servlet-name>AllManufacturers</servlet-name>
+        <servlet-class>core.basesyntax.controller.GetAllManufacturersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>AllManufacturers</servlet-name>
+        <url-pattern>/manufacturer/all</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>CreateManufacturer</servlet-name>
+        <servlet-class>core.basesyntax.controller.CreateManufacturerController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>CreateManufacturer</servlet-name>
+        <url-pattern>/manufacturer/creation</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/creat</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>AllCars</servlet-name>
+        <servlet-class>core.basesyntax.controller.GetAllCarsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>AllCars</servlet-name>
+        <url-pattern>/car/all</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>CreateCar</servlet-name>
+        <servlet-class>core.basesyntax.controller.CreateCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>CreateCar</servlet-name>
+        <url-pattern>/car/creation</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>AddDriverToCar</servlet-name>
+        <servlet-class>core.basesyntax.controller.AddDriverToCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>AddDriverToCar</servlet-name>
+        <url-pattern>/car/driverToCar</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/drivers/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>DeleteDriver</servlet-name>
+        <servlet-class>core.basesyntax.controller.DeleteDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>DeleteDriver</servlet-name>
+        <url-pattern>/driver/delete</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 19]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>DeleteManufacturer</servlet-name>
+        <servlet-class>core.basesyntax.controller.DeleteManufacturerController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>DeleteManufacturer</servlet-name>
+        <url-pattern>/manufacturer/delete</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 20]
```diff
@@ -7,11 +7,110 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>DeleteCar</servlet-name>
+        <servlet-class>core.basesyntax.controller.DeleteCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>DeleteCar</servlet-name>
+        <url-pattern>/car/delete</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/delete</url-pattern>
```

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/controller/AddDriverToCarController.java (hunk 7 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/AddDriverToCarController.java]: Fixed

File: src/main/java/core/basesyntax/controller/AddDriverToCarController.java (hunk 8 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/AddDriverToCarController.java]: Fixed

File: src/main/webapp/web.xml (hunk 12 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/java/core/basesyntax/controller/CreateCarController.java (hunk 9 above)
Comment (vicire) [reply src/main/java/core/basesyntax/controller/CreateCarController.java]: Fixed

File: src/main/webapp/web.xml (hunk 14 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 16 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 17 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 11 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 15 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 18 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 19 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/webapp/web.xml (hunk 20 above)
Comment (vicire) [reply src/main/webapp/web.xml]: Fixed

File: src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java (hunk 10 above)
Comment (vicire) [reply src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java]: I handled this in the table, but sometimes when I add a driver to car, an exception with such trace `cannot add or update a child row a foreign key constraint fails`  occurs
Anyway, I delete this. I test and it adds driver to a car for now

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**